src/org/sosy_lab/common/collect/SkipList.java

```diff
@@ -185,6 +185,14 @@ public SkipList() {
     comparator = (Comparator<? super T>) Comparator.naturalOrder();
   }
 
+  /**
+   * Use the given {@link Random} object for future probabilistic computations.
+   */
+  public void reinitialize(Random pRandom) {
+    Preconditions.checkNotNull(pRandom);
+    randomGenerator = pRandom;
+  }
```

Comment:
I'd like to point to this method so that it's not overseen:
Do you mind such a method?
The point is to enable the use of a controlled Random-object (with a specific seed)
from outside.

Reply:
First, if the `Random` instance is static as it was before, it should probably be a `ThreadLocalRandom`.

Second, which use cases would need changing the random-number generator during the lifetime of a `SkipList` object, and would not be possible if the random-number generator could only be specified at instantation time?

Third, which use cases actually need specifying the random-number generator?
I can only think of cases where users wish to make their code deterministic, e.g., for tests.
However, I see no reason not to make `SkipList` fully deterministic by default.

Thus, I suggest to give each `SkipList` instance its own `Random` instance created with a constant seed.